src/tests/ftest/rebuild/continues_after_stop.py

```diff
@@ -0,0 +1,125 @@
+
+        try:
+            manager.run()
+        except CommandFailure as error:
+            self.log.info(error)
+
+    def test_continuous_after_stop(self):
+        """Verify rebuild continues after one of the ranks is stopped.
+
+        1. Create a pool and a container.
+        2. Run IOR that takes several seconds with a thread.
+        3. After a few seconds, stop one of the ranks (rank 3).
+        4. Look for the start of the rebuild (Rebuild [scanning]) in journalctl with daos_server
+        identifier.
+        5. As soon as the message is detected, stop the rest of the ranks (0, 1, 2).
+        6. Restart the three ranks.
+        7. Wait for rebuild to finish.
+
+        Jira ID: DAOS-6287
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=rebuild
+        :avocado: tags=ContinuesAfterStop,test_continuous_after_stop
+        """
+        # 1. Create a pool and a container.
+        self.log_step("Create a pool and a container.")
+        pool = self.get_pool()
+        container = self.get_container(pool=pool)
+
+        # 2. Run IOR that takes several seconds with a thread.
+        self.log_step("Run IOR that takes several seconds with a thread.")
+        kwargs = {
+            "namespace": "/run/ior/*",
+            "pool": pool,
+            "container": container
+        }
+        thread = threading.Thread(target=self.run_ior_basic, kwargs=kwargs)
+        ior_start_time = journalctl_time()
+        thread.start()
+
+        # 3. After a few seconds, stop one of the ranks (rank 3).
+        self.log_step("After a few seconds, stop one of the ranks (rank 3).")
+        # Wait for IOR to start and write some data. Otherwise rebuild will not occur.
+        time.sleep(5)
+        self.server_managers[0].stop_ranks(ranks=[3], daos_log=self.log)
+
+        # 4. Look for the start of the rebuild (Rebuild [scanning]) in journalctl with daos_server
+        # identifier.
+        msg = ("4. Look for the start of the rebuild (Rebuild [scanning]) in journalctl with "
+               "daos_server identifier.")
+        self.log_step(msg)
+        scanning_found = False
+        for count in range(120):
+            self.log.info("Look for 'Rebuild [scanning]'. Count = %d", count)
+            journalctl_out = get_journalctl(
+                hosts=self.hostlist_servers, since=ior_start_time, until=None,
+                journalctl_type="daos_server")
+            for _, journalctl in enumerate(journalctl_out):
+                data = journalctl["data"]
+                for line in data.splitlines():
+                    if "Rebuild [scanning]" in line:
+                        self.log.info("'Rebuild [scanning]' found: %s", line)
+                        scanning_found = True
+                        break
+                if scanning_found:
+                    break
+            if scanning_found:
+                break
+            time.sleep(1)
```

Comment:
We could utilize this method to handle the retry logic

```
        from general_utils import wait_for_result

        
        def _search():
            journalctl_out = get_journalctl(
                hosts=self.hostlist_servers, since=ior_start_time, until=None,
                journalctl_type="daos_server")

            for _, journalctl in enumerate(journalctl_out):
                data = journalctl["data"]
                for line in data.splitlines():
                    if "Rebuild [scanning]" in line:
                        self.log.info("'Rebuild [scanning]' found: %s", line)
                        return True
            return False

        wait_for_result(self.log, _search, timeout=120, delay=1)
```

Reply:
I updated the code to use `wait_for_result()`. Thanks.